src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -922,22 +906,7 @@ namespace cryptonote
       }
     }
 
-    if (m_service_node_keys)
-    {
-      std::lock_guard<std::mutex> lock{m_quorumnet_init_mutex};
-      // quorumnet_new takes a zmq bind string, e.g. "tcp://1.2.3.4:5678"
-      std::string listen_ip = vm["p2p-bind-ip"].as<std::string>();
-      if (listen_ip.empty())
-        listen_ip = "0.0.0.0";
-      std::string qnet_listen = "tcp://" + listen_ip + ":" + std::to_string(m_quorumnet_port);
-      m_quorumnet_obj = quorumnet_new(*this, qnet_listen);
-    }
-    // Otherwise we may still need quorumnet in remote-only mode, but we construct it on demand
-
-
-#ifdef ENABLE_SYSTEMD
-    sd_notify(0, ("READY=1\nSTATUS=" + get_systemd_status_string(*this)).c_str());
-#endif
+    init_lokimq(vm);
```

Comment:
Currently there's a segfault on startup of a new lokid with no synced blocks yet. The genesis block triggers block added hook and we dereference a null lokimq instance.

```
1  std::thread::joinable                                                                                                                                                                                                                        thread                      154  0x565555bbf50e 
2  lokimq::LokiMQ::set_active_sns                                                                                                                                                                                                               auth.cpp                    83   0x565555bbce0a 
3  cryptonote::core::update_lmq_sns                                                                                                                                                                                                             cryptonote_core.cpp         2071 0x565555873bc4 
4  service_nodes::quorum_cop::block_added                                                                                                                                                                                                       service_node_quorum_cop.cpp 485  0x5655558c9728 
5  cryptonote::Blockchain::handle_block_to_main_chain                                                                                                                                                                                           blockchain.cpp              4189 0x56555583638e 
6  cryptonote::Blockchain::add_new_block                                                                                                                                                                                                        blockchain.cpp              4393 0x56555584689b 
7  cryptonote::Blockchain::init(cryptonote::BlockchainDB *, sqlite3 *, cryptonote::network_type, bool, cryptonote::test_options const *, unsigned long, std::function<epee::span<unsigned char const> const (cryptonote::network_type)> const&) blockchain.cpp              481  0x5655558476d4 
8  cryptonote::core::init(boost::program_options::variables_map const&, cryptonote::test_options const *, std::function<epee::span<unsigned char const> const (cryptonote::network_type)> const&)                                               cryptonote_core.cpp         846  0x565555875fb6 
9  daemonize::daemon::run                                                                                                                                                                                                                       daemon.cpp                  266  0x5655555eceab 
10 daemonizer::daemonize<daemonize::daemon>                                                                                                                                                                                                     posix_daemonizer.inl        87   0x5655556801ae 
11 main                                                                                                                                                                                                                                         main.cpp                    290  0x565555678592 
```

Reply:
`// this feels kinda kludgy to do it this way, but can be looked at later.`

Yup.

Reply:
Moved lokimq initialization before blockchain initialization which should be fine for the genesis block callback.